frontend/src/contexts/Auth.tsx

```diff
@@ -0,0 +1,81 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+
+interface Auth {
+  isLoggedIn?: boolean;
+  logIn: (email: string, password: string) => Promise<Response>;
+  logOut: () => void;
+  getAuthorization: () => string;
+}
+
+const AuthContext = createContext<Auth | undefined>(undefined);
+
+interface Props {
+  children?: React.ReactNode;
+}
+
+interface ErrorResponse {
+  status: false;
+  error: string;
+}
+
+interface SuccessResponse {
+  status: true;
+  name: string;
+  token: string;
+}
+
+type Response = ErrorResponse | SuccessResponse;
+
+export const AuthProvider = ({ children }: Props) => {
+  const [isLoggedIn, setLoggedIn] = useState<boolean>();
+
+  const logIn = async (email: string, password: string) => {
+    const basicAuthToken = getBasicAuth(email, password);
+
+    const request = await fetch('http://localhost:5000/user/login/', {
```

Comment:
Would be nice to have the hostname in the configs

Reply:
I'll look into it.